godot-core/src/docs.rs

```diff
@@ -23,40 +23,56 @@ pub struct StructDocs {
     pub members: &'static str,
 }
 
-/// Created for documentation on
+/// Keeps documentation for Inherent Implementation such as:
 /// ```ignore
 /// #[godot_api]
 /// impl Struct {
 ///     #[func]
 ///     /// This function panics!
 ///     fn panic() -> f32 { panic!() }
+///     #[signal]
+///     /// this signal signals
+///     fn documented_signal(p: Vector3, w: f64);
+///     #[constant]
+///     /// this constant consts
+///     const CON: i64 = 42;
+///
 /// }
 /// ```
 #[derive(Clone, Copy, Debug, Default)]
 pub struct InherentImplDocs {
-    pub methods: &'static str,
-    pub signals: &'static str,
-    pub constants: &'static str,
+    pub methods: Option<&'static str>,
+    pub signals_block: &'static str,
+    pub constants_block: &'static str,
 }
 
 #[derive(Default)]
 struct DocPieces {
     definition: StructDocs,
     inherent: InherentImplDocs,
-    virtual_methods: &'static str,
+    virtual_methods: Option<&'static str>,
 }
 
-#[doc(hidden)]
 /// This function scours the registered plugins to find their documentation pieces,
 /// and strings them together.
 ///
-/// It returns an iterator over XML documents.
+/// Returns an iterator over XML documents.
+///
+/// Documentation for signals and constants is being processed at compile time
+/// and can take the form of an already formatted XML `<block><doc></doc>…</block>`, or an
+/// empty string if no such attribute has been documented.
+///
+/// Since documentation for methods comes from two different sources
+/// -– inherent Implementation (`methods`) and I* trait implementation (`virtual_method_docs`) –-
+/// it is undesirable to merge them at compile time. Instead, they are being kept as an optional
+/// strings of not-yet-parented XML tags
+/// (or nothing if no method has been documented).
```

Comment:
```suggestion
/// Since documentation for methods comes from two different sources
/// -- inherent implementations (`methods`) and `I*` trait implementations (`virtual_method_docs`) --
/// it is undesirable to merge them at compile time. Instead, they are being kept as an optional
/// strings of not-yet-parented XML tags (or nothing if no method has been documented).
```